packages/block-library/src/navigation-link/edit.js

```diff
@@ -242,6 +243,37 @@ export default function NavigationLinkEdit( {
 	const itemLabelPlaceholder = __( 'Add link…' );
 	const ref = useRef();
 
+	const isPostType =
+		( kind && 'post-type' === kind ) || ( type && 'post' === type );
+
+	const hasId = id && ! isNaN( id );
+	const postStatus = useSelect( ( select ) => {
+		if ( isPostType ) {
+			const { getEntityRecord } = select( coreStore );
+			const entityRecord = getEntityRecord( 'postType', type, id );
+
+			return entityRecord?.status;
+		}
+
+		return null;
+	} );
+
+	// Check Navigation Link validity if:
+	// 1. Link is 'post-type'.
+	// 2. It has an id.
+	// 3. It's neither null, nor undefined, as valid items might be either of those while loading.
+	// If those conditions are met, check if
+	// 1. The post status is published.
+	// 2. The Navigation Link item has no label.
+	// If either of those is true, invalidate.
+	const isInvalid =
```

Comment:
This feels like it could be a bit more pleasant to the eyes 😄 
Maybe something like that:
```
const isInvalid =
	isPostType &&
	hasId &&
	postStatus &&
	( postStatus !== 'publish' || ! label );
```
Didn't give it much thought, so it could become better.

Reply:
I don't think it gets much better than that.
I removed the label check though, as it doesn't make sense in the editor — it would, for instance, make it impossible to delete the label on a page link in order to edit it.